src/basilisp/lang/seq.py

```diff
@@ -91,57 +91,6 @@ def with_meta(self, meta: Optional[IPersistentMap]) -> "Cons[T]":
         return Cons(self._first, seq=self._rest, meta=meta)
 
 
-class _Sequence(IWithMeta, ISequential, ISeq[T]):
```

Comment:
`_Sequence` types aren't really needed since we can define the `sequence` function on any `Iterable` below in terms of `LazySeq` instead.